azure-pipelines.yml

```diff
@@ -205,23 +205,23 @@ stages:
     - script: |
         set -e
         mkdir build && cd build
-        cmake .. -DOSX_PACKAGE=ON -DPYTHON_BINDINGS=ON -DWITH_EXAMPLES=ON -DWITH_SERIAL_BACKEND=OFF -DWITH_ZSTD=OFF
+        cmake .. -Werror=dev -DCOMPILE_WARNING_AS_ERROR=ON -DCOMPILE_WARNING_AS_ERROR=ON -DOSX_PACKAGE=ON -DPYTHON_BINDINGS=ON -DWITH_EXAMPLES=ON -DWITH_SERIAL_BACKEND=OFF -DWITH_ZSTD=OFF
```

Comment:
You pass it twice here.

Also, why do you pass `-Werror=dev` to CMake? Does it do anything?

Reply:
I will fix the twice - that was a typo.

the `-Werror=dev` causes any `cmake` warnings to be flagged as errors, and halts/throws an error. (as opposed to the `-DCOMPILE_WARNING_AS_ERROR=ON` which controls the compiler)

From the cmake man page:
```
-Werror=<what>
Treat CMake warnings as errors. <what> must be one of the following:

dev
      Make developer warnings errors.
      Make warnings that are meant for the author of the CMakeLists.txt files errors. 
      By default this will also turn on deprecated warnings as errors.
deprecated
      Make deprecated macro and function warnings errors.
      Make warnings for usage of deprecated macros and functions, that are meant for 
      the author of the CMakeLists.txt files, errors.
```

There is no way (that I'm aware) to do this inside the Cmake file itself.

and again - something for libiio developers, but not end users.